CTOR-880: New Pack(storage::hp::alletra::restapi) (#4475)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/hardware-storage-hpe-alletra-restapi.md

---
id: hardware-storage-hpe-alletra-restapi
title: Alletra
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Compatibility

This connector is designed to be compatible with the following products.

| Product     | Model        | Versions |
|-------------|--------------|----------|

> | HPE Alletra | 9000         | NA       |

Reviewer: ```suggestion
| HPE Alletra | 9000         | N/A       |
```

## Connector dependencies

The following monitoring connectors will be installed when you install the **HPE Alletra REST API** connector through the

> **Configuration > Monitoring Connector Manager** menu:

Reviewer: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu:
```

* [Base Pack](./base-generic.md)

## Pack assets

### Templates

The Monitoring Connector **HPE Alletra REST API** brings a host template:

* **HW-Storage-HPE-Alletra-API-custom**

The connector brings the following service templates (sorted by the host template they are attached to):

<Tabs groupId="sync">
<TabItem value="HW-Storage-HPE-Alletra-API-custom" label="HW-Storage-HPE-Alletra-API-custom">

| Service Alias | Service Template                                    | Service Description                   | Discovery  |
|:--------------|:----------------------------------------------------|:--------------------------------------|:----------:|
| Capacity      | HW-Storage-HPE-Alletra-Capacity-Restapi-custom      | Check capacity by storage type        |            |
| Disk-Status   | HW-Storage-HPE-Alletra-Disk-Status-Restapi-custom   | Check the state of the physical disks | X          |

> | Disk-Usage    | HW-Storage-HPE-Alletra-Disk-Usage-Restapi-custom    | Check the disk usage                  | X          |

Reviewer: ```suggestion
| Disk-Usage    | HW-Storage-HPE-Alletra-Disk-Usage-Restapi-custom    | Check disk usage                  | X          |
```

| Licenses      | HW-Storage-HPE-Alletra-Licenses-Restapi-custom      | Check the state of the licenses       |            |
| Volume-Status | HW-Storage-HPE-Alletra-Volume-Status-Restapi-custom | Check the status of the volumes       | X          |

> | Volume-Usage  | HW-Storage-HPE-Alletra-Volume-Usage-Restapi-custom  | Check the volume usage                | X          |

Reviewer: ```suggestion
| Volume-Usage  | HW-Storage-HPE-Alletra-Volume-Usage-Restapi-custom  | Check volume usage                | X          |
```

> | Volume-Usage  | HW-Storage-HPE-Alletra-Volume-Usage-Restapi-custom  | Check volume usage                | X          |

Reviewer 2: ```suggestion
| Capacity      | HW-Storage-HPE-Alletra-Capacity-Restapi-custom      | Check capacity by storage type        |            |
| Disk-Status   | HW-Storage-HPE-Alletra-Disk-Status-Restapi-custom   | Check the state of the physical disks | X          |
| Licenses      | HW-Storage-HPE-Alletra-Licenses-Restapi-custom      | Check the state of the licenses       |            |
| Volume-Status | HW-Storage-HPE-Alletra-Volume-Status-Restapi-custom | Check the status of the volumes       | X          |
```

> The services listed above are created automatically when the **HW-Storage-HPE-Alletra-API-custom** host template is used.

> If **Discovery** is checked, it means a service discovery rule exists for this service template.

</TabItem>
<TabItem value="Not attached to a host template" label="Not attached to a host template">

| Service Alias | Service Template                                   | Service Description    | Discovery  |
|:--------------|:---------------------------------------------------|:-----------------------|:----------:|

> | Disk-Usage    | HW-Storage-HPE-Alletra-Disk-Usage-RESTAPI-custom   | Check the disk usage   | X          |

Reviewer: ```suggestion
| Disk-Usage    | HW-Storage-HPE-Alletra-Disk-Usage-RESTAPI-custom   | Check disk usage   | X          |
```

> | Volume-Usage  | HW-Storage-HPE-Alletra-Volume-Usage-RESTAPI-custom | Check the volume usage | X          |

Reviewer: ```suggestion
| Volume-Usage  | HW-Storage-HPE-Alletra-Volume-Usage-RESTAPI-custom | Check volume usage | X          |
```

> The services listed above are not created automatically when a host template is applied. To use them, [create a service manually](/docs/monitoring/basic-objects/services), then apply the service template you want.

> If **Discovery** is checked, it means a service discovery rule exists for this service template.

</TabItem>
</Tabs>

### Discovery rules

#### Service discovery

| Rule name                                           | Description                                               |
|:----------------------------------------------------|:----------------------------------------------------------|

> | HW-Storage-HPE-Alletra-Restapi-Disk-Status-Id       | Discover the physical disks and monitor their status.     | 

Reviewer: same as FR

| *disk_name*#disk.space.free.bytes       | B     |
| *disk_name*#disk.space.usage.percentage | %     |

</TabItem>
<TabItem value="Licenses" label="Licenses">

| Name                                            | Unit  |
|:------------------------------------------------|:------|
| licenses.total.count                            | count |
| licenses.expired.count                          | count |
| *license_expiration*#license.expiration.seconds | s     |

</TabItem>
<TabItem value="Volume-Status" label="Volume-Status">

| Name                   | Unit  |
|:-----------------------|:------|
| volumes.total.count    | count |
| volumes.normal.count   | count |
| volumes.degraded.count | count |
| volumes.failed.count   | count |
| volumes.unknown.count  | count |
| *volume_id*#status     | N/A   |

</TabItem>
<TabItem value="Volume-Usage" label="Volume-Usage">

| Name                                        | Unit  |
|:--------------------------------------------|:------|
| *volume_name*#volume.space.usage.bytes      | B     |
| *volume_name*#volume.space.free.bytes       | B     |
| *volume_name*#volume.space.usage.percentage | %     |

</TabItem>
</Tabs>

## Prerequisites

> he following procedure is a excerpt from the official [HPE documentation](https://support.hpe.com/hpesc/public/docDisplay?docId=a00118592en_us&page=GUID-77CF1ECF-98D5-44E1-B040-F54F17374A20.html).

Reviewer: ```suggestion
The following procedure is an excerpt from the official [HPE documentation](https://support.hpe.com/hpesc/public/docDisplay?docId=a00118592en_us&page=GUID-77CF1ECF-98D5-44E1-B040-F54F17374A20.html).
```

1. Log in to the CLI as `Super`, `Service`, or any role granted the `wsapi_set` right.
2. Start the WSAPI server.
    ```
   cli%
   startwsapi
   ```
3. To configure WSAPI, enter `setwsapi` in the CLI.

> Please refer to the official documentation for the details.

Reviewer: ```suggestion
Please refer to the official documentation for more details.
```

## Installing the monitoring connector

### Pack

1. If the platform uses an *online* license, you can skip the package installation
instruction below as it is not required to have the connector displayed within the

> **Configuration > Monitoring Connector Manager** menu.

Reviewer: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu.
```

If the platform uses an *offline* license, install the package on the **central server**
with the command corresponding to the operating system's package manager:

<Tabs groupId="sync">
<TabItem value="Alma / RHEL / Oracle Linux 8" label="Alma / RHEL / Oracle Linux 8">

```bash
dnf install centreon-pack-hardware-storage-hpe-alletra-restapi
```

</TabItem>
<TabItem value="Alma / RHEL / Oracle Linux 9" label="Alma / RHEL / Oracle Linux 9">

```bash
dnf install centreon-pack-hardware-storage-hpe-alletra-restapi
```

</TabItem>
<TabItem value="Debian 11 & 12" label="Debian 11 & 12">

```bash
apt install centreon-pack-hardware-storage-hpe-alletra-restapi
```

</TabItem>
<TabItem value="CentOS 7" label="CentOS 7">

```bash
yum install centreon-pack-hardware-storage-hpe-alletra-restapi
```

</TabItem>
</Tabs>

2. Whatever the license type (*online* or *offline*), install the **HPE Alletra REST API** connector through

> the **Configuration > Monitoring Connector Manager** menu.

Reviewer: ```suggestion
the **Configuration > Connectors > Monitoring Connectors** menu.
```

| WARNINGDISKSUNKNOWN   | Threshold                                                                                                                                                                                                                                                                                                                                                                                  |                                             |             |
| CRITICALDISKSUNKNOWN  | Threshold                                                                                                                                                                                                                                                                                                                                                                                  |                                             |             |

> | WARNINGSTATUS         | Threshold                                                                                                                                                                                                                                                                                                                                                                                  | %\{status\} =~ /^(new\|degraded\|unknown)$/ |             |
> | CRITICALSTATUS        | Threshold                                                                                                                                                                                                                                                                                                                                                                                  | %\{status\} =~ /failed/                     |             |

Reviewer 2: ```suggestion
| WARNINGSTATUS         | Define the condition to match for the returned status to be WARNING                                                                                                                                                                                                                                                                                                                                                                   | %\{status\} =~ /^(new\|degraded\|unknown)$/ |             |
| CRITICALSTATUS        | Define the condition to match for the returned status to be CRITICAL                                                                                                                                                                                                                                                                                                                                                                    | %\{status\} =~ /failed/                     |             |
```
